### DISCUSSION TITLE
CoreWF and MAUI Blazor

### DISCUSSION THREAD

**Discussion by jcmontoya** [UiPath/corewf#206] [Q&A]
I created a test MAUI Blazor project and added a simple workflow with one C# expression.  The windows version works with no issues. The Andorid version works without C# Expressions but when I add a C# expression and enable CompileExpressions I get the following error: 

System.Resources.MissingSatelliteAssemblyException: 'The satellite assembly named "System.Activities.resources.dll, PublicKeyToken=null" for fallback culture "en" either could not be found or could not be loaded. This is generally a setup problem. Please consider reinstalling or repairing the application.'

The error happens in the bold line.

```
private async void  Test() {
        try
        {
            var stream = await FileSystem.OpenAppPackageFileAsync("Workflow1.jmc");
            var reader = new StreamReader(stream);
            var xamlWFString = reader.ReadToEnd();

            IDictionary<String, object> properties = new Dictionary<string, object>();
            properties.Add("x", "Test");

            ActivityXamlServicesSettings settings = new ActivityXamlServicesSettings
            {
                CompileExpressions = true
            };

            //Error in this line:            
            DynamicActivity helloWorldActivity = (DynamicActivity)ActivityXamlServices.Load(new StringReader(xamlWFString.ToString()), settings);
            var result = System.Activities.WorkflowInvoker.Invoke(helloWorldActivity, properties);
            string final = result["res"] as String;
        }
        catch (Exception ex)
        {
            string msg = ex.Message; 
        }

    }
```
Test Workflow:

![image](https://user-images.githubusercontent.com/6794769/165642975-3529afd5-0426-4230-bdc0-7677d0ab6967.png)

Any help would be appreciated.

**Comment by dmetzgar** [comment]
In the folder containing the application executable and DLLs, is there a subfolder with the name "en"? It sounds like the expression compilation is trying to get a localized string and not finding the resources.

**Reply by dmetzgar** [reply]
Part of the problem too is that we do this: https://github.com/UiPath/CoreWF/blob/3a66a486ad7aed1a8537a87662ba5f000d6cdfee/src/UiPath.Workflow.Runtime/AssemblyInfo.cs#L20-L20

The normal fallback location is in the main assembly but this forces it to a satellite assembly (see [here](https://docs.microsoft.com/en-us/dotnet/core/extensions/package-and-deploy-resources#net-core-resource-fallback-process)). I think that could easily be changed. It would allow your program to get past that point but if you need localized strings you may need to figure out how to package the satellite assemblies for Android.
